extension/src/environment.rs

```diff
@@ -168,14 +171,14 @@ pub trait Ext {
 	type AccountId;
 
 	/// Returns a reference to the account id of the current contract.
-	fn address(&self) -> &Self::AccountId;
+	fn address(&self) -> Self::AccountId;
 }
 
 impl Ext for () {
 	type AccountId = ();
 
-	fn address(&self) -> &Self::AccountId {
-		&()
+	fn address(&self) -> Self::AccountId {
```

Comment:
This was changed just to test if it works. Reason why I don't add the method `caller()` because it will bring back stuffs that @evilrobot-01 removed before https://github.com/r0gue-io/pop-node/pull/265 that removed the `Config` associated type. The `caller()` method returns `OriginFor<Config>` so if we want that method, we need to revert the changed Frank made.

Reply:
To be clear, I'm not against adding stuff that is needed, but a useful thing to consider is that taking a trait bound on the whole Config trait makes mocking overly complex. An example is needing to add a mock pallet-contracts impl just to do basic things.

Only expose what is required, nothing more. It may be easier, but has consequences downstream for implementors.